### PR TITLE
Upgrade from Chromium 100 to Chromium 101.

### PR DESCRIPTION
**Pull request by mkarolin**
Fixes brave/brave-browser#21507

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by cdesouza-chromium — APPROVED** [review]
LGTM

File: build/ios/mojom/mojom_objc_generator.py [hunk 1]
```diff
@@ -270,7 +270,8 @@ class BaseDictionaryValueMojoTypemap(MojoTypemap):
     @staticmethod
     def IsMojoType(kind):
         return (mojom.IsStructKind(kind) and
-                kind.qualified_name == 'mojo_base.mojom.DictionaryValue')
+                (kind.qualified_name ==
+                 'mojo_base.mojom.DeprecatedDictionaryValue'))
```
Comment (kylehickinson) [inline build/ios/mojom/mojom_objc_generator.py:274]: Are there any usages of DictionaryValue still? I assume if so then this will no longer work for those

**Review by samartnik — APPROVED** [review]
++

**Review by tmancey — APPROVED** [review]
LGTM++

**Review by kylehickinson — APPROVED** [review]
iOS++

File: browser/ui/webui/settings/brave_wallet_handler.cc [hunk 2]
```diff
@@ -170,53 +170,3 @@ void BraveWalletHandler::SetActiveNetwork(const base::Value::List& args) {
...
-
-// Remove this method and pass it the non-Deprecated version instead to
-// RegisterMessageCallback once CL 3483819 lands (in Chromium 101).
-void BraveWalletHandler::GetCustomNetworksListDeprecated(
-    base::Value::ConstListView args) {
-  base::Value::List args_list;
-  for (const auto& arg : args)
-    args_list.Append(arg.Clone());
-  GetCustomNetworksList(args_list);
-}
```
Comment (darkdh) [inline browser/ui/webui/settings/brave_wallet_handler.cc:222]: shouldn't we also remove these functions from `brave/browser/ui/webui/settings/brave_wallet_handler.h`?

**Review by emerick — APPROVED** [review]
Rewards changes LGTM

File: browser/ui/webui/settings/brave_wallet_handler.cc (hunk 2 above)
Comment (mkarolin) [reply browser/ui/webui/settings/brave_wallet_handler.cc:222]: Yes, yes we should 😅 Fixed in 7203bb7a2ef6d9f6479e2c77be31d6af6bf7f7ed

**Review by darkdh — APPROVED** [review]
wallet ++

**Review by simonhong — APPROVED** [review]
browser/themes ++
browser/ui/views/inforbars/brave_wayback_* ++
browser/widevine ++

**Review by szilardszaloki — APPROVED** [review]
Rewards changes LGTM.

**Review by goodov — APPROVED** [review]
lgtm! haven't found anything suspicious.

**Review by bsclifton — APPROVED** [review]
LGTM!

**Review by iefremov — APPROVED** [review]

**Comment by mkarolin** [comment]
Force-pushed the branch: rebased on master to resolve conflicts.

File: components/de_amp/browser/test/de_amp_browsertest.cc [hunk 3]
```diff
@@ -53,7 +58,18 @@ const char kTestNonAmpBody[] =
...
   DeAmpBrowserTest() {
-    feature_list_.InitAndEnableFeature(de_amp::features::kBraveDeAMP);
+    std::vector<base::Feature> disabled_features = {};
+#if BUILDFLAG(IS_MAC)
+    // On Mac, the DeAmpBrowserTest.AmpURLNotStoredInHistory test crashes
+    // due to https://crbug.com/1284500: DCHECK in
+    // blink::ContentToVisibleTimeReporter::TabWasShown when BFCache is
+    // used. To get around the crash, disabling BFCache for these tests
+    // until the upstream bug is fixed.
+    disabled_features.push_back({features::kBackForwardCache});
```
Comment (bridiver) [inline components/de_amp/browser/test/de_amp_browsertest.cc:68]: how are we tracking this so we know when we can remove this?

**Review by bridiver — APPROVED** [review]

File: components/de_amp/browser/test/de_amp_browsertest.cc (hunk 3 above)
Comment (mkarolin) [reply components/de_amp/browser/test/de_amp_browsertest.cc:68]: I'll an issue to track this.

Comment (mkarolin) [reply components/de_amp/browser/test/de_amp_browsertest.cc:68]: Created https://github.com/brave/brave-browser/issues/22347

**Comment by mkarolin** [comment]
CI:
* noplatform failed due to a known audit issue unrelated to the PR
* Windows failed due to an intermittently failing upstream browser test - ok to ignore.

**Comment by mkarolin** [comment]
**When uplifting, also uplift:**

- [ ] https://github.com/brave/brave-browser/issues/22431

**Pull request merged**